src/main.cpp

```diff
@@ -8,24 +8,38 @@
 #include <iostream>
 #include <vector>
 #include "Box.hpp"
-#include "shared_pointer.hpp"
 #include "unique_pointer.hpp"
+#include "shared_pointer.hpp"
 
 int main(int argc, const char * argv[]) {
-    UniquePointer<int> ptr1;
-    UniquePointer<int>* ptr2 = new UniquePointer<int>();
-    UniquePointer<int> ptr3 = make_unique<int>();
-    std::vector<UniquePointer<int> > testArray;
+    // UniquePointer<int> ptr1;
```

Comment:
Add proper unit tests for both shared ptr and unique ptr using assert.... Create two independent helper function for these and call them in main.

Reply:
done